### ISSUE TITLE
unnecessary_lambda_linter should work with explicit return()

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1703]
Split off from #1567, since handling `return()` as well is another layer of complication

Relevant test cases:

```
 test_that("cases with braces are caught", {
   linter <- unnecessary_lambda_linter()
  
   expect_lint(
     "lapply(x, function(xi) { return(print(xi)) })",
     "xxx",
     linter
   )
   
   expect_lint(
     trim_some("
       lapply(x, function(xi) {
         return(print(xi))
       })
     "),
     "xxx",
     linter
   )
 })
```

_Originally posted by @MichaelChirico in https://github.com/r-lib/lintr/issues/1567#issuecomment-1260456007_
